src/commonMain/kotlin/exchange.dydx.abacus/calculator/MarginModeCalculator.kt

```diff
@@ -126,48 +131,54 @@ internal object MarginModeCalculator {
             return subaccountNumber
         }
         val marketId = parser.asString(tradeInput?.get("marketId")) ?: return subaccountNumber
-        val subaccounts = parser.asMap(account?.get("subaccounts")) ?: return subaccountNumber
-
-        var lastSubaccountNumber = subaccountNumber
-        for ((_, item) in subaccounts) {
-            val subaccount = parser.asMap(item) ?: continue
-            val childSubaccountNumber =
-                parser.asInt(subaccount["subaccountNumber"]) ?: subaccountNumber
-
-            if (childSubaccountNumber % NUM_PARENT_SUBACCOUNTS == subaccountNumber) {
-                if (containsMarket(parser, subaccount, marketId)) {
-                    return childSubaccountNumber
-                } else {
-                    if (childSubaccountNumber > lastSubaccountNumber) {
-                        lastSubaccountNumber = childSubaccountNumber
-                    }
-                }
+        val subaccounts = parser.asNativeMap(account?.get("subaccounts")) ?: return subaccountNumber
+
+        val utilizedSubaccountsMarketIdMap = subaccounts.mapValues {
```

Comment:
i've seen this calculated in a few places, should we just expose this as a helper function?